### ISSUE TITLE
ImageSource is now open for inheritance but the getUri method is still final and cannot override it

### ISSUE THREAD

**Issue by Daniel3711997** [facebook/react-native#46150]
### Description

ImageSource is now open for inheritance but the getUri method is still final and cannot override it

### Steps to reproduce

1. Install the [react-native-fast-image](https://github.com/DylanVann/react-native-fast-image) package
2. Use it in the application
3. Try to compile the app for android

### React Native Version

0.75.2

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
❯ npx react-native info
info Fetching system and libraries information...
System:
  OS: macOS 14.6.1
  CPU: (12) arm64 Apple M3 Pro
  Memory: 481.47 MB / 18.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.16.0
    path: ~/.nvm/versions/node/v20.16.0/bin/node
  Yarn: Not Found
  npm:
    version: 10.8.2
    path: ~/.nvm/versions/node/v20.16.0/bin/npm
  Watchman:
    version: 2024.08.19.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /Users/user/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "33"
      - "34"
    Build Tools:
      - 33.0.0
      - 34.0.0
    System Images:
      - android-33 | ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
      - android-33 | Google Play ARM 64 v8a
      - android-34 | ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2024.1 AI-241.18034.62.2411.12169540
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.12
    path: /Users/user/.sdkman/candidates/java/current/bin/javac
  Ruby:
    version: 2.7.5
    path: /Users/user/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.2
    wanted: 0.75.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: false
```

### Stacktrace or Logs

```text
/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageSource.java:72: error: isResource() in FastImageSource cannot override isResource() in ImageSource
public boolean isResource() {
^
overridden method is final

/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageSource.java:101: error: getUri() in FastImageSource cannot override getUri() in ImageSource
public Uri getUri() {
^
overridden method is final
```

### Reproducer

Just follow the Steps to reproduce

**Comment by shubhamguptadream11** [comment]
Fixed it here: https://github.com/facebook/react-native/pull/46152

**Comment by cortinico** [comment]
This is fixed, will be shipped in the upcoming 0.75.x version

**Comment by kbqdev** [comment]
[react-native+0.75.2.patch](https://github.com/user-attachments/files/16862980/react-native%2B0.75.2.patch)
File patch-package if anyone need
